### DISCUSSION TITLE
Local authenticator initial password reset not working with 2FA

### DISCUSSION THREAD

**Discussion by kschoenberger** [requarks/wiki#3969] [Error / Bug Report]
Hi all,

I'm running WikiJS as docker service with the image requarks/wiki:2.5.

What I configured:

- Enforce 2FA for all users.
- Added a new user to local authenticator and checked the setting "Require password change on first login".

What I except:
1. User is logging in with initial password.
2. 2FA setup.
3. Password change.
4. User is logged in with new password.

What happens:
1. User is logging in with initial password.
2. 2FA setup. Entering the first time the 2FA code from the authenticator app, the page moves on to the password change, but doesn't finish the 2FA setup. Entering another time the code results in an error. Therefore it is not possible to change the password.

Workaround:
1. Reload page.
2. Login once more.
3. Enter 2FA code. --> Page forwards to password change, but the input form for 2FA code still is present.
4. Enter 2FA code once more. --> Error appears and input form closes.
5. Change password.
6. Get finally redirected to the wiki.

If someone has already a solution for or has troubles to reproduce, please let me know.

BR,
Klaus

**Comment by Programmierschmied** [comment]
Problem still exists.

Sorry for the duplicate at: https://github.com/requarks/wiki/discussions/4274

**Comment by felix-exon** [comment]
![image](https://user-images.githubusercontent.com/24978665/185193375-596b19e3-fd16-46dd-8a54-cfa938acba5f.png)

**Comment by cpath-star** [comment]
I'm still seeing this issue
